app/src/main/java/com/microsoft/device/samples/dualscreenexperience/presentation/history/ui/OrderDetails.kt

```diff
@@ -0,0 +1,293 @@
+    }
+
+    // Calculate padding for LazyColumn
+    val paddingValues = with(LocalDensity.current) {
+        PaddingValues(bottom = 20.dp + topBarPadding.toDp() + bottomNavPadding.toDp())
+    }
+
+    val columnModifier = if (isLandscape)
+        Modifier
+            .fillMaxWidth(0.915f)
+            .padding(top = 32.dp)
+            .fillMaxHeight()
+    else
+        Modifier
+            .fillMaxWidth(0.9f)
+            .fillMaxHeight()
+
+    var showDialog by remember { mutableStateOf(false) }
+    val updateShowDialog = { newValue: Boolean -> showDialog = newValue }
+    var selectedOrderItem: OrderItem? by remember { mutableStateOf(null) }
+    val updateOrderItem = { newItem: OrderItem -> selectedOrderItem = newItem }
+
+    Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.TopCenter) {
+        Column(
+            modifier = columnModifier,
+        ) {
+            OrderHeader(order, showTwoPages, isSmallWidth)
+            Spacer(modifier = Modifier.height(22.dp))
+            OrderItems(order.items, paddingValues, isExpanded, isSmallWidth, updateShowDialog, updateOrderItem)
+        }
+        if (showDialog)
+            AddToOrderDialog(
+                selectedOrderItem,
+                updateShowDialog,
+                getProductFromOrderItem,
+                addToOrder,
+                isSmallHeight
+            )
+    }
+}
+
+@Composable
+fun OrderHeader(order: Order, showTwoPages: Boolean, isSmallWidth: Boolean) {
+    if (showTwoPages) {
+        Text(
+            text = stringResource(R.string.toolbar_history_details_title),
+            style = MaterialTheme.typography.h4,
+            color = MaterialTheme.colors.onBackground
+        )
+        Spacer(Modifier.height(4.dp))
+    }
+
+    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
+        OrderDate(
+            modifier = Modifier.weight(1f),
+            orderTimestamp = order.orderTimestamp,
+            isSmallWidth = isSmallWidth
+        )
+        Column(
+            modifier = Modifier.weight(1f),
+            verticalArrangement = spacedBy(4.dp),
+            horizontalAlignment = Alignment.End
+        ) {
+            OrderId(orderId = order.orderId, isSmallWidth = isSmallWidth)
+            OrderAmount(orderPrice = order.totalPrice, isSmallWidth = isSmallWidth)
+        }
+    }
+}
+
+@Composable
+fun OrderItems(
+    orderItems: MutableList<OrderItem>,
+    paddingValues: PaddingValues,
+    isExpanded: Boolean,
+    isSmallWidth: Boolean,
+    updateShowDialog: (Boolean) -> Unit,
+    updateOrderItem: (OrderItem) -> Unit
+) {
+    LazyColumn(
+        verticalArrangement = spacedBy(25.dp),
+        contentPadding = paddingValues
+    ) {
+        orderItems.map { orderItem ->
+            item {
+                OrderItemPreview(orderItem, isExpanded, isSmallWidth, updateShowDialog, updateOrderItem)
+            }
+        }
+    }
+}
+
+@Composable
+fun OrderItemPreview(
```

Comment:
Having a composable named "preview" when it is used in the layout itself and not in the jetpack compose preview might be confusing for other developers

Reply:
ah good point, I'll update that!